packages/entity-server/src/routes/hierarchy/middleware/filter.ts

```diff
@@ -58,35 +68,56 @@ const operatorToSqlComparator = {
   '==': '=' as const, // Exact match
   '!=': '!=' as const, // Does not match
   '=@': 'ilike' as const, // Contains sub string
+  '<': '<' as const, // Less than
+  '<=': '<=' as const, // Less than or equal
+  '>': '>' as const, // Greater than
+  '>=': '>=' as const, // Greater than or equal
 };
 type Operator = keyof typeof operatorToSqlComparator;
 
 const filterOperators = Object.keys(operatorToSqlComparator) as Operator[];
 
-const formatComparisonValue = (value: string | string[], operator: Operator) => {
+type Value = string | string[] | number | number[];
+
+const formatComparisonValue = (value: Value, operator: Operator) => {
   if (operator === '=@' && typeof value === 'string') {
     return `%${value}%`;
   }
 
   return value;
 };
 
-const convertValueToAdvancedCriteria = (operator: Operator, value: string | string[]) => {
+const formatValue = <T extends keyof EntityFilterQuery>(field: T, value: string) =>
+  isNumericField(field) ? parseFloat(value) : value;
+
+const convertValueToAdvancedCriteria = (
+  field: keyof EntityFilterQuery,
+  operator: Operator,
+  value: string,
+) => {
+  const formattedValue = value.includes(MULTIPLE_VALUES_DELIMITER)
```

Comment:
@rohan-bes  I had to move value parsing from `extractFilterFromQuery()` to this method. The reason is that here the field is `keyof EntityFilterQuery`, e.g. `attributes_type`, and not yet formatted to `attributes->>type`. We generally use `_` to define the filterable fields and their types.

This change has a bit of a risk, but luckily `entity-server` has an amazing test suite, which includes some filter tests too! I can also do some manual testing for this one